### DISCUSSION TITLE
Custom context values using ATOTI

### DISCUSSION THREAD

**Discussion by ditruter** [atoti/atoti#515] [General]
My organization's Active Pivot cube has some custom context values defined that must be informed in order to launch queries on it.

When I use the UI I can build the MDX query and also set the custom context values, however when I try to do the same from ATOTI I can't find any way to assign the custom context values.

Is there any way to do it?

**Comment by bloussou** [comment]
Hello,
Did you try using the [`cube.shared_context`](https://docs.atoti.io/latest/lib/atoti.cube.html#atoti.cube.Cube.shared_context) ?

**Reply by ditruter** [reply]
Thanks, that was my first option, but i think cube.shared_context only works when i am creating a new cube into my session. 

I have tried the intuitive way:

cube = mysession.cubes("organization cube")

But it returns me the type QueryCube who doesn´t have the property shared_context. I think there is no way to get the type Cube when trying to select a preexisting cube, only when i am creating my own cube into my session.

So i am stuck.

**Reply by bloussou** [reply]
Hello,
I have opened this feature request to add the possibility to add context to queries easily : https://github.com/atoti/atoti/issues/517

**Reply by ditruter** [reply]
Thank you! 

We are a very large corporation in Spain. We are starting to work with Active Pivot in some projects and this little detail is a big great stopper for us.

Thanks again!

**Reply by bloussou** [reply]
Hello,
This will be available in the next release

**Comment by bloussou** [comment]
Hello,
From `0.6.5` you can use the *context* parameter of the [`query()`](https://docs.atoti.io/0.6.5/lib/atoti/atoti.cube.html#atoti.Cube.query) method.

**Comment by ditruter** [comment]
I saw it last friday and it works perfect!

Thank you very much!
